scripts/ioc_deploy.py

```diff
@@ -0,0 +1,327 @@
+    )
+    parser.add_argument(
+        "--name",
+        "-n",
+        action="store",
+        default="",
+        help="The name of the repository to deploy. This is a required argument. If it does not exist on github, we'll also try prepending with 'ioc-common-'.",
+    )
+    parser.add_argument(
+        "--release",
+        "-r",
+        action="store",
+        default="",
+        help="The version of the IOC to deploy. This is a required argument.",
+    )
+    parser.add_argument(
+        "--ioc-dir",
+        "-i",
+        action="store",
+        default=current_default_target,
+        help=f"The directory to deploy IOCs in. This defaults to $EPICS_SITE_TOP/ioc, or {EPICS_SITE_TOP_DEFAULT}/ioc if the environment variable is not set. With your current environment variables, this defaults to {current_default_target}.",
+    )
+    parser.add_argument(
+        "--github_org",
+        "--org",
+        action="store",
+        default=current_default_org,
+        help=f"The github org to deploy IOCs from. This defaults to $GITHUB_ORG, or {GITHUB_ORG_DEFAULT} if the environment variable is not set. With your current environment variables, this defaults to {current_default_org}.",
+    )
+    parser.add_argument(
+        "--auto-confirm",
+        "--confirm",
+        "--yes",
+        "-y",
+        action="store_true",
+        help="Skip the confirmation promps, automatically saying yes to each one.",
+    )
+    parser.add_argument(
+        "--dry-run",
+        action="store_true",
+        help="Do not deploy anything, just print what would have been done.",
+    )
+    parser.add_argument(
+        "--verbose",
+        "-v",
+        "--debug",
+        action="store_true",
+        help="Display additional debug information.",
+    )
+    return parser
+
+
+class ReturnCode(enum.IntEnum):
+    SUCCESS = 0
+    EXCEPTION = 1
+    NO_CONFIRM = 2
+
+
+def main(args: CliArgs) -> int:
+    """
+    All main steps of the script.
+
+    Will either return an int return code or raise.
+    """
+    if not (args.name and args.release):
+        raise ValueError(
+            "Must provide both --name and --release. Check ioc-deploy --help for usage."
+        )
+
+    logger.info("Running ioc-deploy: checking inputs")
+    upd_name = finalize_name(name=args.name, github_org=args.github_org)
+    upd_rel = finalize_tag(
+        name=upd_name, github_org=args.github_org, release=args.release
+    )
+    deploy_dir = get_target_dir(name=upd_name, ioc_dir=args.ioc_dir, release=upd_rel)
+
+    logger.info(f"Deploying {args.github_org}/{upd_name} at {upd_rel} to {deploy_dir}")
+    if not args.auto_confirm:
+        user_text = input("Confirm release source and target? y/n\n")
+        if not user_text.strip().lower().startswith("y"):
+            return ReturnCode.NO_CONFIRM
+    if Path(deploy_dir).exists():
+        logger.info(f"{deploy_dir} exists, skip git clone step.")
```

Comment:
I don't recall what `epics-release` does in this case, but it seems to me that if we're trying to deploy a release to a directory that already exists, we should just bail out. Is there a use case here that I'm not thinking of?

Reply:
You might be right. I also can't remember what `epics-release` does. The intent was to allow us to rebuild the ioc but I see now that the script is starting to mix verbs- it doesn't do a "full deploy", it does a "clone if needed, make if possible" which could cross some wires...

Reply:
Yeah, after turning this around in my head for a bit, I'm fully on board. Let's bail out if there's "something" already there.

Reply:
New output from re-running a build, exits prior to asking for confirmation:
```
$ ./scripts/ioc-deploy --name ads-ioc -r 0.6.2 --ioc-dir /cds/home/z/zlentz/ioc
INFO     ioc-deploy: Running ioc-deploy: checking inputs
WARNING  ioc-deploy: ads-ioc is not an ioc name, trying ioc-common-ads-ioc
INFO     ioc-deploy: ioc-common-ads-ioc exists in pcdshub
WARNING  ioc-deploy: Did not find release 0.6.2 in pcdshub/ioc-common-ads-ioc
INFO     ioc-deploy: Release R0.6.2 exists in pcdshub/ioc-common-ads-ioc
INFO     ioc-deploy: Deploying pcdshub/ioc-common-ads-ioc at R0.6.2 to /cds/home/z/zlentz/ioc/common/ads-ioc/R0.6.2
ERROR    ioc-deploy: Deploy directory /cds/home/z/zlentz/ioc/common/ads-ioc/R0.6.2 already exists! Aborting.
```